_source/_posts/2020-12-18-how-to-use-webauthn-csharp-dotnet.md

```diff
@@ -0,0 +1,780 @@
+
+        await UpdateCounter(user, credential, result.Counter);
+
+        var claims = new List<Claim>
+        {
+        new Claim(ClaimTypes.Name, user.Profile.Email)
+        };
+
+        var claimsIdentity = new ClaimsIdentity(claims, CookieAuthenticationDefaults.AuthenticationScheme);
+
+        await HttpContext.SignInAsync(CookieAuthenticationDefaults.AuthenticationScheme, new ClaimsPrincipal(claimsIdentity));
+
+        return Json(result);
+    }
+    catch (Exception e)
+    {
+        return Json(new AssertionVerificationResult { Status = "error", ErrorMessage = e.Message });
+    }
+}
+```
+
+Run the project again, click the **Sign In** in the menu, enter the email you signed up with, and click the **Sign In** button. You will see a prompt like this:
+
+{% img blog/webauthn-csharp/image11.png alt:"" width:"800" %}{: .center-image }
+
+Swipe your finger, and you will be signed in and redirected to the profile page. To show the current user profile data, I will use Okta .NET SDK to get the current user details and display them.
+
+Add a `Profile` method with the following code:
+
+```cs
+public async Task<IActionResult> Profile()
+{
+  var subject = HttpContext.User.Claims.First(claim => claim.Type == ClaimTypes.Name).Value;
+  var user = await oktaClient.Users.GetUserAsync(subject);
+  return View(user);
+}
+```
+
+Next, create the `Profile.cshtml` file in the `Views\Account` folder that looks like this:
+
+```html
+@model Okta.Sdk.IUser
+
+@{
+  ViewData["Title"] = "User Profile";
+}
+
+<h1>User Profile</h1>
+
+<div>
+    <dl class="row">
+    <dt class = "col-sm-2">
+        @Html.DisplayNameFor(model => model.Profile.FirstName)
+    </dt>
+    <dd class = "col-sm-10">
+        @Html.DisplayFor(model => model.Profile.FirstName)
+    </dd>
+    <dt class = "col-sm-2">
+        @Html.DisplayNameFor(model => model.Profile.LastName)
+    </dt>
+    <dd class = "col-sm-10">
+        @Html.DisplayFor(model => model.Profile.LastName)
+    </dd>
+    <dt class = "col-sm-2">
+        @Html.DisplayNameFor(model => model.Profile.Email)
+    </dt>
+    <dd class = "col-sm-10">
+        @Html.DisplayFor(model => model.Profile.Email)
+    </dd>
+    </dl>
+</div>
+<div>
+    <a asp-action="EditProfile">Edit Profile</a>
+</div>
+```
+
+The view renders profile details such as the user's first name and last name, email, city, and country code.
+
+Rerun the app, click the **Profile** button, and you will see the details of the currently logged in user.
+
+That's it! You can register and sign with passwordless authentication and view the details of the current user.
+
+In this tutorial, we learned how to use WebAuthn in ASP.NET Core and Okta to make passwordless authentication more secure.
+
+Give passwordless authentication a try, get the [full source code of the project from GitHub](https://github.com/oktadeveloper/OktaWebAuthn) and let us know what you think in the comments below.
```

Comment:
We try to prefix our blog post repos with "okta" and suffix them with "example". Also, it's a good idea (for SEO) to include the keywords in the repo name. I'd recommend renaming this repo to "okta-dotnet-webauthn-example". Please update the README to include instructions so the user can clone, configure for Okta, and run.